src/generate.js

```diff
@@ -258,81 +270,71 @@ export const generateCSSRuleset = (
     const originalElements = {...handledDeclarations.elements};
 
     // NOTE(emily): This mutates handledDeclarations.elements.
-    const prefixedDeclarations = prefixAll(handledDeclarations.elements);
+    const prefixedElements = prefixAll(handledDeclarations.elements);
 
-    const prefixedRules = flatten(
-        objectToPairs(prefixedDeclarations).map(([key, value]) => {
-            if (Array.isArray(value)) {
-                // inline-style-prefixer returns an array when there should be
-                // multiple rules for the same key. Here we flatten to multiple
-                // pairs with the same key.
-                return value.map(v => [key, v]);
-            }
-            return [[key, value]];
-        })
-    );
-
-    // Calculate the order that we want to each element in `prefixedRules` to
-    // be in, based on its index in the original key ordering.
-    const sortOrder = {};
-    for (let i = 0; i < handledDeclarations.keyOrder.length; i++) {
-        const key = handledDeclarations.keyOrder[i];
-        sortOrder[key] = i;
-
-        // In order to keep most prefixed versions of keys in about the same
-        // order that the original keys were in but placed before the
-        // unprefixed version, we generate the prefixed forms of the keys and
-        // set their order to the same as the original key minus a little bit.
-        const capitalizedKey = `${key[0].toUpperCase()}${key.slice(1)}`;
-        const prefixedKeys = [
-            `Webkit${capitalizedKey}`,
-            `Moz${capitalizedKey}`,
-            `ms${capitalizedKey}`,
-        ];
-        for (let j = 0; j < prefixedKeys.length; ++j) {
-            if (!originalElements.hasOwnProperty(prefixedKeys[j])) {
-                sortOrder[prefixedKeys[j]] = i - 0.5;
-                originalElements[prefixedKeys[j]] = originalElements[key];
-            }
-        }
-    }
+    const elementNames = Object.keys(prefixedElements);
+    if (elementNames.length !== handledDeclarations.keyOrder.length) {
+        // There are some prefixed values, so we need to figure out how to sort
+        // them.
+        //
+        // Loop through prefixedElements, looking for anything that is not in
+        // sortOrder, which means it was added by prefixAll. This means that we
+        // need to figure out where it should appear in the sortOrder.
+        for (let i = 0; i < elementNames.length; i++) {
+            if (!originalElements.hasOwnProperty(elementNames[i])) {
+                // This element is not in the sortOrder, which means it is a prefixed
+                // value that was added by prefixAll. Let's try to figure out where it
+                // goes.
+                let originalStyle;
+                if (elementNames[i][0] === 'W') {
```

Comment:
I just looked through the code of inline-style-prefixer and I was unable to find any others. @rofrischmann can you confirm?